provision/README.md

```diff
@@ -1,43 +1,141 @@
-# PROVISION service
+# Provision service
 
-PROVISION service provides an HTTP API to interact with Mainflux.
+Provision service provides an HTTP API to interact with [Mainflux][mainflux]. 
+Provision service is used to setup initial applications configuration i.e. things, channels, connections and certificates that will be required for the specific use case especially useful for gateway provision. 
+For gateways to communicate with [Mainflux][mainflux] connection params are required (mqtt host, thing, channels, certificates...). To get the connection parameters gateway will send a request to [Bootstrap](../bootstrap/README.md) service providing `<external_id>` and `<external_key>` in request. To make a request to [Bootstrap](../bootstrap/README.md) service you can use [Agent](https://github.com/mainflux/agent) service on a gateway. To create bootstrap configuration you can use [Bootstrap](../bootstrap/README.md) or `Provision` service. [Mainflux UI](https://github.com/mainflux/ui) uses [Bootstrap](../bootstrap/README.md) service for creating gateway configurations.  `Provision` service should provide an easy way of provisioning your gateways i.e creating bootstrap configuration and as many things and channels that your setup requires. Also you may use provision service to create certificates for each thing. Each service running on gateway may require more than one thing and channel for communication. Let's say that you are using services [Agent](https://github.com/mainflux/agent) and [Export](https://github.com/mainflux/export) on a gateway, if you enabled mtls each service will need a thing and certificate for access to [Mainflux][mainflux] and you will need two channels for `Agent` (`data` and `control`) and one for `Export` or you could use as many as you find suitable, this kind of setup we can call `provision layout`.
```

Comment:
Maybe use `configuration` instead of connection params. Maybe also something like this: 
```
To create a bootstrap configuration, you can use Bootstrap service API directly (bootstrap/README.md), or you can use UI or Provision service.
```